E2E Framework: do not load warm cookies for Pre-Release Tests.

worldomonation opened the pull request:
## Proposed Changes

This PR changes the behavior of the cookie preloading in the framework.

Because the Pre-Release Tests don't benefit much (if at all) from preloading warm cookies, this PR makes changes to disable the process. 

Key changes:
- change `NODE_CONFIG_ENV` to 'pre-release' for Pre-Release Tests.
- add a clause to short-circuit if the `NODE_CONFIG_ENV` matches "pre-release".

## Testing Instructions

Ensure the following build configurations are passing:
  - [x] Pre-Release E2E

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

worldomonation committed:
.teamcity/_self/projects/WebApp.kt
- change NODE_CONFIG_ENV to 'pre-release' for Pre-Release Tests.

packages/calypso-e2e/src/jest-playwright-config/global-setup.ts
- add clause to short-circuit if the suite being run is Pre-Release Tests

WunderBart commented on packages/calypso-e2e/src/jest-playwright-config/global-setup.ts:
```diff
@@ -7,6 +7,12 @@ import pwConfig from './playwright-config';
 export default async (): Promise< void > => {
 	const { AUTHENTICATE_ACCOUNTS } = envVariables;
 
+	// If running in the Pre-Release Tests environment in CI,
+	// don't execute the cookie refresh.
+	if ( process.env.NODE_CONFIG_ENV === 'pre-release' ) {
+		return;
+	}
```
Since `JETPACK_TARGET` is included in the [`envVariables`](https://github.com/Automattic/wp-calypso/blob/12e689757062860c8a62a58c2017b1de85986e2e/packages/calypso-e2e/src/env-variables.ts#L13-L30), should we also add this env there for consistency?

worldomonation replied:
@WunderBart I chose not to add the `NODE_CONFIG_ENV` to the framework `envVariables` because this value isn't really exclusive to the E2E framework, being something [defined by Node by default](https://github.com/node-config/node-config/wiki/Environment-Variables). I didn't think it was right to co-opt this value by restricting the range of values it can take. 

If the thought is we should, where possible, explicitly define the environment variables that we use in the E2E framework, then I think it would be better to rework the existing `AUTHENTICATE_ACCOUNTS` to default to nothing instead, and make the necessary tweaks to support that. 

What do you think?

WunderBart replied:
> I think it would be better to rework the existing AUTHENTICATE_ACCOUNTS to default to nothing instead, and make the necessary tweaks to support that.

This sounds reasonable to me. I think being explicit regarding the pre-auth accounts would bring more clarity to the process. We could also try nullifying the env var by passing `AUTHENTICATE_ACCOUNTS=''`, but I'm not sure it wouldn't require additional tweaks as well.

youbrokesomething commented:
@worldomonation Are there any performance changes that we should expect from this change?

worldomonation committed:
.teamcity/_self/projects/WebApp.kt
- revert changes.

packages/calypso-e2e/src/env-variables.ts
- set default value to an empty array.

packages/calypso-e2e/src/jest-playwright-config/global-setup.ts
- change the clause condition to check for <= 0

worldomonation committed:
Update list of accounts to pre-authenticate for other builds.

worldomonation commented:
> @worldomonation Are there any performance changes that we should expect from this change?

Pre-Release Tests may be slightly faster, but the improvements are probably consumed by a couple of tests needing to authenticate now instead.

The main change resulting from this PR is that we should no longer see random failures on Pre-Release Tests caused by the hanging process of cookie warming, which does happen once in a while.

worldomonation commented on packages/calypso-e2e/src/env-variables.ts:
```diff
@@ -19,7 +19,7 @@ const defaultEnvVariables: SupportedEnvVariables = {
 	GUTENBERG_NIGHTLY: false,
 	COBLOCKS_EDGE: false,
-	AUTHENTICATE_ACCOUNTS: [ 'simpleSitePersonalPlanUser', 'atomicUser', 'defaultUser' ],
+	AUTHENTICATE_ACCOUNTS: [],
```
This was the only change required, thankfully.

worldomonation commented on packages/calypso-e2e/src/jest-playwright-config/global-setup.ts:
```diff
@@ -13,29 +13,33 @@ export default async (): Promise< void > => {
-		await browser.close();
+	// If the list of accounts for which to pre-authenticate and save cookies
+	// for is empty, then don't run.
+	if ( AUTHENTICATE_ACCOUNTS.length <= 0 ) {
```
Changing the conditional to reject early.

WunderBart replied:
nit: since array length cannot be negative:

```suggestion
	if ( AUTHENTICATE_ACCOUNTS.length === 0 ) {
```

worldomonation replied:
Commited suggestion, thanks for the nit.

WunderBart reviewed (approved):
Just one _nit_ comment. LGTM! 🚀

dpasque reviewed (approved):
Also LGTM! Thanks for doing this! 🚀

worldomonation committed:
Update equality check

- array cannot be negative

Co-authored-by: Bart Kalisz <bartlomiej.kalisz@gmail.com>
